### PR TITLE
fix(copilot): setup M.state when not initialized

### PR DESCRIPTION
**Pull request by ethanh20009**
fixes https://github.com/yetone/avante.nvim/issues/861

Copilot was broken and wouldn't work at all for me. PR resolves it.

### CODE REVIEW

File: lua/avante/providers/copilot.lua [hunk 1]
```diff
@@ -85,7 +85,7 @@ H.chat_completion_url = function(base_url) return Utils.trim(base_url, { prefix
 local M = {}
 
 H.refresh_token = function()
-  if not M.state then error("internal initialization error") end
+  if not M.state then M.setup() end
```
Comment (aarnphm) [inline lua/avante/providers/copilot.lua:88]: ugh this should be setup before hand, not to be called here.

**Review by aarnphm — CHANGES_REQUESTED** [review]
I don't think this fix it. let's investigate

File: lua/avante/providers/copilot.lua (hunk 1 above)
Comment (ethanh20009) [reply lua/avante/providers/copilot.lua]: Found the root cause:
Copilot does not have an api_key_name as is setup in other plugin. 
In provider init, it checks for no api_key_name and early returns. In our case this is wrong as we need copilot setup.

So added a bool for providers that are set up without api keys.

**Comment by aarnphm** [comment]
cc @yetone

**Comment by errnoh** [comment]
Tested this with Copilot (gpt-4o-2024-08-06) and this fixes #861 for me. ( running avante https://github.com/yetone/avante.nvim/commit/9891b0365607373553b5dab74494a5e81b01e3c2 )

**Comment by yetone** [comment]
‌‌‌‌Hello, I fixed this issue in this way because I really couldn't find a reason not to set up the provider: https://github.com/yetone/avante.nvim/pull/868

**Pull request closed without merging**